src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,224 @@
+      `zoom` variable is used to keep track of the seat map is in a zoomed state or not. This variable is used to
+    decide whether to show a + or - button.
+    `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <h1>Seat map</h1>
+    <!-- You can implement a seat map by using `amp-list`, `amp-bind` and `amp-pan-zoom` where
+    `amp-list` loads the list of seats. 
+    We are using the `transformEnd` action from [amp-pan-zoom](https://www.ampproject.org/docs/reference/components/amp-pan-zoom)
+    to update the value of transform.scale variable.
+    We display 2 types of seats: wheelchair and standard. Each seat can be available, unavailable and selected. Everytime a seat is 
+    selected we add the seat id to the `seatmap.selectedSeats` variable.
+    -->
+     <div class="list-container">
+     <amp-list layout="fill" src="/json/seats.json" items="." single-item noloading>
+      <div placeholder>
+        <svg class="opaque disabled-seat seat-map-svg" viewBox="0 0 [[.Width]] [[.Height]]" preserveAspectRatio="xMidYMin slice">
+          [[range .Seats]] 
+          [[if .Wheelchair]]             
+            <rect x="[[.X]]" y="[[.Y]]" width="[[.Width]]" height="[[.Height]]" class="special-seat" rx="[[.Rx]]" ry="[[.Ry]]"/>
+              <image class="special-seat-image"
+                x="[[.X]]" y="[[.Y]]" width="[[.Width]]" height="[[.Height]]"
+                      xlink:href="/img/accessible-unavailable-24px.svg"
+                      tabindex="0"
+                      role="button"></image>
+          [[end]]
+          [[if .Standard]]  
+           <rect class="seat unavailable" 
+                           x="[[.X]]" y="[[.Y]]" width="[[.Width]]" height="[[.Height]]"
+                          role="button"
+                          tabindex="0" rx="[[.Rx]]" ry="[[.Ry]]"/>   
+            [[end]]
+            [[end]]
+              
+        </svg>
+      </div>
+          <template type="amp-mustache">
+          <amp-pan-zoom layout="fill" id="pan-zoom"
+            on="transformEnd:AMP.setState({seatmap: {scale: event.scale, x: event.x, y: event.y, zoom: event.scale <= 1 ? 0 : 1}})">
+            <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+                {{#seats}}                
+                {{#standard }}
+                    <rect class="seat selected"
+                          hidden [hidden]="!seatmap.selectedSeats.includes('{{id}}')"
+                          x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}"
+                          on="tap:AMP.setState({
+                            seatmap: {selectedSeats: copyAndSplice(seatmap.selectedSeats, seatmap.selectedSeats.indexOf('{{id}}'), 1)}
```

Comment:
You can use Filip's hint feature to explain more details inside the code section.

Reply:
good idea!